warehouse/query-core/src/main/java/datawave/query/attributes/SummarySize.java

```diff
@@ -0,0 +1,215 @@
+package datawave.query.attributes;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import datawave.query.Constants;
+import datawave.query.postprocessing.tf.PhraseIndexes;
+
+/**
+ * Represents options for a summary that have been specified within an #SUMMARY_SIZE function. An instance of {@link SummarySize} can easily be captured as a
+ * parameter string using {@link SummarySize#toString()}, and transformed back into a {@link SummarySize} instance via {@link SummarySize#from(String)}.
+ */
+public class SummarySize implements Serializable {
+
+    private static final long serialVersionUID = 6769159729743311079L;
+
+    private static final int DEFAULT_SIZE = 150;
+    private static final Logger log = LoggerFactory.getLogger(SummarySize.class);
+
+    private int summarySize;
+    private ArrayList<String> contentNamesList;
+    private boolean only;
+
+    public SummarySize() {
+        summarySize = 0;
```

Comment:
This is a bit confusing. The default summary size is 150, but the default constructor sets the size to zero.

Reply:
currently the default size is used if the input to the function is misformed